sdk/storage/Azure.Storage.DataMovement/MigrationGuide.md

```diff
@@ -52,6 +55,79 @@ dotnet add package Azure.Storage.DataMovement.Blobs
+It is no longer a static type, though **we recommend maintaining a singleton instance** for optimal usage.
+
+All transfer methods for upload, download, and copy, for both files and directories, have been replaced with a single instance method:
+```csharp
+Task<TransferOperation> StartTransferAsync(
+    StorageResource sourceResource,
+    StorageResource destinationResource,
+    TransferOptions transferOptions = default,
+    CancellationToken cancellationToken = default)
+```
+The appropriate upload, download, or copy operation will be performed based on the `StorageResource` instances passed. E.g., a `sourceResource` pointing to a local file and a `destinationResource` pointing to an Azure blob will perform an upload. More information on `StorageResource` can be found in the following section on [storage resources and providers](#storage-resources-and-providers).
+
+The following table is a quick reference of where to find functionality from legacy `TransferManager` APIs. This **does not** imply fully identical behavior between libraries.
+
+| Operation | Legacy `TransferManager` API | Modern API
+| --------- | ---------- | ---------- |
+| Upload, Download, Copy | `UploadAsync`, `UploadDirectoryAsync`, `DownloadAsync`, `DownloadDirectoryAsync`, `CopyAsync`, `CopyDirectoryAsync` | `TransferManager.StartTransferAsync`
+| Limit concurrent transfers | `Configurations.ParallelOperations` | `TransferManagerOptions.MaximumConcurrency` (applied at TransferManager construction)
+| Limit concurrent listings | `Configurations.MaxListingConcurrency` | Not implemented
+| Limit memory usage | `Configurations.MaximumCacheSize` | Not implemented
+| Set chunking size | `Configurations.BlockSize` | `TransferOptions.MaximumTransferChunkSize` (applied at TransferManager construction)
+| Alter user agent string | `Configurations.UserAgentPrefix` | Not implemented
+
+#### Storage Resources and Providers
+
+The modern library introduces a new and unified `StorageResource` type. While the legacy library directly used types like `CloudBlob` and `CloudFile` in its API, the modern library uses `StorageResource` to define transfer source and destination. Storage resources can be further divided into two subtypes: **items** and **containers**. Examples of items are block blobs and local files. Examples of containers are blob containers and local folders.
+
+**Important:** a transfer **must** be between two items or two containers. An exception will be thrown when attempting to transfer an item to a container and vice versa.
+
+`StorageResource` instances are obtained through providers. Providers are often scoped to a single storage service and will have unique APIs to acquire `StorageResource` instances as well as properly authenticate them. Here is an example using providers to create an upload to an Azure blob. Further examples can be found in our [migration samples](#migration-samples).
+
+```csharp
+LocalFilesStorageResourceProvider files = new();
```

Comment:
Just FYI, Amanda has a PR out to remove this constructor and turn `LocalFilesStorageResourceProvider` methods static.